packages/rocketchat-lib/server/lib/notifyUsersOnMessage.js

```diff
@@ -100,9 +100,10 @@ function notifyUsersOnMessage(message, room) {
 		}
 	}
 
-	// Update all the room activity tracker fields
+	// Update all the room activity tracker fields, including livechat-analytics lm timestamps
 	// This method take so long to execute on gient rooms cuz it will trugger the cache rebuild for the releations of that room
-	RocketChat.models.Rooms.incMsgCountAndSetLastMessageById(message.rid, 1, message.ts, RocketChat.settings.get('Store_Last_Message') && message);
+	RocketChat.models.Rooms.incMsgCountAndSetLastMessageById(room, message.token, 1, message.ts, RocketChat.settings.get('Store_Last_Message') && message);
```

Comment:
As I said below, you don't need to change the method `incMsgCountAndSetLastMessageById` to implement your code.
You have created a new file listening `afterSaveMessage` callback, why didn't you implement the code there?
https://github.com/RocketChat/Rocket.Chat/pull/11238/files#diff-9943cbbd0d16d3c0c7306a507793042dR1